tests/platform_tests/test_power_budget_info.py

```diff
@@ -0,0 +1,60 @@
+import logging
+import re
+import json
+import pytest
+from tests.common.helpers.assertions import pytest_assert
+from cli.util import get_skip_mod_list
+
+logger = logging.getLogger('__name__')
+
+pytestmark = [
+    pytest.mark.topology('t2')
+]
+
+
+def test_power_redis_db(duthosts, enum_supervisor_dut_hostname, tbinfo):
+    """
+    @summary: verify the output for power budget policy using
+    redis command for chassis
+    checks for each psu the supplied power
+    checks consumed power for each present module
+    """
+    logger.info("verifying redis dump for power budget")
+    duthost = duthosts[enum_supervisor_dut_hostname]
+    skip_mod_list = get_skip_mod_list(duthost)
+    exp_total_supp_power = 0
+    exp_total_cons_power = 0
+
+    redis_out = duthost.command("redis-dump -d 6 -y -k \"*power*\"")
+    out_dict = json.loads(redis_out['stdout'])
+    power_budget = out_dict.keys()
+    
+    for pb_name in power_budget:
+        for out_val in out_dict[pb_name]['value']:
+            if re.match('Supplied Power', out_val):
+                n_psu = (re.split('Supplied Power ', out_val))[1]
+                if n_psu not in skip_mod_list:
+                    sup_power = float(out_dict[pb_name]['value'][out_val])
+                    pytest_assert(sup_power > 0,
+                                  "expected supplied power for psu {} is expected to 0 or less".format(n_psu))
```

Comment:
psu {} is expected to be > 0 ? 
And can it print the sup_power value as well ?

Reply:
not psu {} but supplied power for psu expected to be greater than 0